lsコマンドを作る２(-aオプション付きのlsコマンド)のls.rbプログラムを作成

a-kuroki-gs opened the pull request.

a-kuroki-gs committed:
lsコマンドを作る２(-aオプション付きのlsコマンド)のls.rbプログラムを作成

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -80,10 +34,57 @@ def print_elements(element)
+  dir_all = target_dir_all.map(&:sort)
+else
+  normal_dir_all = []
+  target_dir_all.each do |target_dir|
+    normal_dir = []
+    target_dir.each do |target|
+      normal_dir << target unless target[0] == '.'
+    end
+    normal_dir_all << normal_dir
+  end
```
`select` か `reject` を使えば簡単にできそうな気がしますが、どうでしょうか。
https://docs.ruby-lang.org/ja/latest/method/Array/i/filter.html
https://docs.ruby-lang.org/ja/latest/method/Array/i/reject.html

a-kuroki-gs replied:
rejectメソッドを用いて簡単に表記しました！

yoshitsugu reviewed (commented):
コメントしました！

a-kuroki-gs committed:
lsコマンド2の課題でメソッドの変更などの修正を行いました

.から始まるものを除外する操作でrejectメソッドを用いた
昇順に並べる操作をaオプションのありなしに関わらず行う
表示する要素がないときにreturnでメソッドから抜けるように変更

yoshitsugu commented on 04.ls/ls.rb:
```diff
@@ -73,17 +27,61 @@ def print_element_array(dir, max_width_array)
+if params[:a].nil?
+  target_dir_all =
+    target_dir_all.map do |target_dir|
+      target_dir.reject { |target| target[0] == '.' }
```
「最初が `.` ではじまっていないかどうか？」は  `start_with?` を使うと直感的かもです。
https://docs.ruby-lang.org/ja/latest/method/String/i/start_with=3f.html

a-kuroki-gs replied:
「使えるメソッドがないか探してみる」ってとても大事ですね。
勉強になりました(^^ゞ
`start_with?`を使うことにします！

yoshitsugu reviewed (approved):
OKです。一点だけコメントしました。

a-kuroki-gs committed:
直感的に分かりやすいメソッドを用いて変更しました

. から始まるか調べる操作で、start_with?を使いました
